packages/shared/src/contexts/payment/index.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import type { ReactElement } from 'react';
+import dynamic from 'next/dynamic';
+import { isIOSNative } from '../../lib/func';
+import { PaddleSubProvider } from './Paddle';
+import type { PaymentContextProviderProps } from './context';
+
+const StoreKitSubProvider = dynamic(() =>
+  import('./StoreKit').then((mod) => mod.StoreKitSubProvider),
+);
+
+export const PaymentContextProvider = ({
```

Comment:
From where do you use this <PaymentContextProvider> now?

Just to check so we have one wrapping context which is PaymentContextProvider
Then it just renders <StoreKit or <Paddle inside which expose the same output right?
Just pass drill the same twice?

We just need to make sure they keep aligned then.

Reply:
> From where do you use this now?
Same places as before.

> Just to check so we have one wrapping context which is PaymentContextProvider
Then it just renders <StoreKit or <Paddle inside which expose the same output right?

Previously PaymentContextProvider was just the paddle stuff. Now it either returns Paddle or StoreKit.

> We just need to make sure they keep aligned then.

They both return the same expected output. Only difference is that in StoreKit, it returns null/undefined for the gifting related parts, so to satisfy the type interface of the Payment context

Reply:
Yeah the last one is totally fine, just need to ensure adding new stuff forces additions on both sides.
But think you do that with shared types already.